frontend/packages/console-dynamic-plugin-sdk/src/api/api-types.ts

```diff
@@ -43,3 +44,31 @@ export type UseK8sWatchResources = <R extends ResourcesObject>(
 export type UseResolvedExtensions = <E extends Extension>(
   ...typeGuards: ExtensionTypeGuard<E>[]
 ) => [ResolvedExtension<E>[], boolean, any[]];
+
+/* Horizontal Nav Types */
+export type PageComponentProps<R extends K8sResourceCommon = K8sResourceCommon> = {
+  filters?: any;
+  selected?: any;
+  match?: any;
+  obj?: R;
+  params?: any;
+  customData?: any;
+  showTitle?: boolean;
+  fieldSelector?: string;
+};
+
+export type NavPage = {
+  href?: string;
+  name?: string;
+  component?: React.ComponentType<PageComponentProps>;
+  // use this to cascade routes under the page
+  exact?: boolean;
+};
+
+export type HorizontalNavFacadeProps = {
+  className?: string;
+  obj?: { loaded: boolean; data: K8sResourceCommon };
```

Comment:
Is this necessary or should we put the onus on the individual page do to its own fetching as horizontal nav isn't always for a resource page.